src/clarity-angular/popover/popover.directive.ts

```diff
@@ -16,24 +16,28 @@ export class PopoverDirective {
     constructor(private templateRef: TemplateRef<any>, private viewContainer: ViewContainerRef) {
     }
 
-    // TODO: current implementation only allows a single popover to be open at any given time
-    // will need to revisit this when the requirements change for future components such as nested menu
     @Input() set clrPopover(open: boolean) {
         if (open) {
-            if (openCount === 0) {
+            if (this.popoverOptions.allowMultipleOpen) {
                 this.createPopover();
             } else {
-                waiting.push(() => {
+                if (openCount === 0) {
                     this.createPopover();
-                });
+                } else {
+                    waiting.push(() => {
+                        this.createPopover();
+                    });
+                }
             }
         } else {
             this.viewContainer.clear();
             this.destroyPopover();
```

Comment:
As per discussion with @youdz and @reddolan we are moving to an implementation where we close all popovers on scroll.